lib/logtail/log_entry.rb

```diff
@@ -127,12 +128,12 @@ def convert_to_runtime_context(frame)
       end
 
       def logtail_frame?(frame)
-        return false if frame.absolute_path.nil? || current_gem_path.nil?
-        frame.absolute_path.start_with?(current_gem_path)
+        return false if frame.absolute_path.nil? || logtail_gem_paths.empty?
+        frame.absolute_path.start_with?(logtail_gem_paths)
       end
 
-      def current_gem_path
-        @current_gem_path ||= $LOAD_PATH.select { |path| __FILE__.start_with?(path) }.max_by(&:length)
+      def logtail_gem_paths
+        @logtail_gem_paths ||= $LOAD_PATH.select { |path| path.match(LOGTAIL_GEM_REGEX) }
```

Comment:
:lipstick: 
```suggestion
        @logtail_gem_paths ||= $LOAD_PATH.select { |path| path.match?(LOGTAIL_GEM_REGEX) }
```

Reply:
I had to revert this, `match?` is not available in older ruby versions.